cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -234,17 +237,22 @@ class InstanceArchitectureCompatibilityValidator(Validator):
     Verify that head node and compute instance types imply compatible architectures.
     """
 
-    def _validate(self, instance_type, architecture: str):
+    def _validate(self, instance_type_info_list: List[InstanceTypeInfo], architecture: str):
         head_node_architecture = architecture
-        compute_architectures = AWSApi.instance().ec2.get_supported_architectures(instance_type)
-        if head_node_architecture not in compute_architectures:
-            self._add_failure(
-                "The specified compute instance type ({0}) supports the architectures {1}, none of which are "
-                "compatible with the architecture supported by the head node instance type ({2}).".format(
-                    instance_type, compute_architectures, head_node_architecture
-                ),
-                FailureLevel.ERROR,
-            )
+
+        compute_architectures = set()
```

Comment:
the `set` is not needed because you need to compare head_node to each_compute resource and not head_node to the incremental set of compute_resources